src/flash/display/MovieClip.ts

```diff
@@ -145,16 +171,36 @@ module Shumway.AVMX.AS.flash.display {
       var eventClass = this.sec.flash.events.Event.axClass;
       var queue: MovieClip[] = movieClipClass._callQueue;
       movieClipClass._callQueue = [];
+      var unsortedScripts: FrameScript [] = [];
+      
       for (var i = 0; i < queue.length; i++) {
         var instance = queue[i];
 
-        if (instance._hasFlags(DisplayObjectFlags.NeedsLoadEvent)) {
-          instance._removeFlags(DisplayObjectFlags.NeedsLoadEvent);
-          instance.dispatchEvent(eventClass.getInstance(events.Event.AVM1_LOAD));
+        if (movieClipClass.frameNavigationModel === FrameNavigationModel.SWF1) {
+          // AVM1 action blocks must be executed exactly in the same order they appear in the SWF
+          // file. We keep track of the position for each action block at parsing time and carry that
+          // information on to their wrapping functions (via the precedence property). Here we queue
+          // up all such functions so we can sort and execute them after this loop.
+          if (instance._hasFlags(DisplayObjectFlags.NeedsLoadEvent)) {
+            instance._removeFlags(DisplayObjectFlags.NeedsLoadEvent);
+            release || assert(instance._symbol);
+            var handler = (<AVM1.Lib.AVM1SymbolBase>
+              AVM1.Lib.getAVM1Object(instance, instance._symbol.avm1Context))._onLoadHandler;
```

Comment:
We have to carry the precedence somehow, which is set as a property on the function `_onLoadHandler` (which implements the `FrameScript` interface) refers to. Retrieving that information from event handlers would result in a way uglier hack.